Backport patches for rdma cm patches

Backport patch for ip_check_addr() for kernels v2.6.22/23/24

Signed-of-by: David Wilder <dwilder@us.ibm.com>
---
diff -Naurp ofa_kernel.orig/drivers/infiniband/core/addr.c ofa_kernel/drivers/infiniband/core/addr.c
--- ofa_kernel.orig/drivers/infiniband/core/addr.c	2009-12-07 15:42:33.000000000 -0600
+++ ofa_kernel/drivers/infiniband/core/addr.c	2009-12-07 15:53:18.000000000 -0600
@@ -130,8 +130,7 @@ int rdma_translate_ip(struct sockaddr *a
 	case AF_INET6:
 		read_lock(&dev_base_lock);
 		for_each_netdev(&init_net, dev) {
-			if (ipv6_chk_addr(&init_net,
-					  &((struct sockaddr_in6 *) addr)->sin6_addr,
+			if (ipv6_chk_addr(&((struct sockaddr_in6 *) addr)->sin6_addr,
 					  dev, 1)) {
 				ret = rdma_copy_addr(dev_addr, dev, NULL);
 				break;


